source/kraken/fill_disruption_from_chaos.cpp

```diff
@@ -79,6 +81,21 @@ make_cause(const chaos::Cause& chaos_cause, nt::new_disruption::DisruptionHolder
 
 }
 
+//return the period of activity for a day
+static boost::posix_time::time_period activity_period(const boost::gregorian::date& date,
+                                                          const boost::posix_time::time_period& period){
+    auto day_period = bt::time_period(bt::ptime(date, bt::hours(0)), bt::hours(24));
+    //we get the validity period of the imapct for this day
+    return day_period.intersection(period);
+}
+
+//return the time period of circulation of a vj for one day
+static boost::posix_time::time_period execution_period(const boost::gregorian::date& date,
+                                                          const nt::VehicleJourney& vj){
+    return bt::time_period(bt::ptime(date, bt::seconds(vj.stop_time_list.front().departure_time)),
+            bt::ptime(date, bt::seconds(vj.stop_time_list.back().arrival_time)));
```

Comment:
I think you should add an epsilon to the arrival date, the end in not in the time_period